### ISSUE TITLE
Link-in-text-block problems

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3774]
The recent changes to link-in-text-block rule has a few problems. First, the `data` is not converting the colors into hex, so the failureSummary is using `[Object object]` instead of the hex color. Also, the contrast is not being truncated so the contrast is very long decimal number.

> "Fix any of the following:
  The link has insufficient color contrast of 1.6620953314177012:1 with the surrounding text. (Minimum contrast is 3:1, link text: [object Object], surrounding text: [object Object])
  The link has no styling (such as underline) to distinguish it from the surrounding text"

Lastly, the issue is reporting false positives for the following HTML:

```html
<p>hello world, goodbye world
  <a href="#" style="text-decoration: none; background: #eee;">
    <style>a { color: #333 }</style>
    <img src="https://randomuser.me/api/portraits/men/54.jpg" alt=""/>
  </a>
</p>
```

The matches function uses `textContent` to determine if the link has text, but `textContext` grabs the text of the style element so it passes the matches.
